test/routes/home_route_test.dart

```diff
@@ -69,4 +72,75 @@ void main() {
       expect(ContentControl.instance.state.allSongs.songs, [songWith()]);
     });
   });
+
+  testWidgets('app shows exit confirmation toast', (WidgetTester tester) async {
+    await Prefs.confirmExitingWithBackButton.set(true);
+    final ToastObserver toastObserver = ToastObserver(tester);
+    final AppObserver appObserver = AppObserver(tester);
+    await tester.runAppTest(() async {
+      expect(
+        await appObserver.haveCloseRequest(
+          () => toastObserver.expectShowsToast(
+            BackButtonInterceptor.popRoute,
+            message: l10n.pressOnceAgainToExit,
+            reason: 'Expected the app to ask for confirmation before exiting',
+          ),
+        ),
+        isFalse,
+        reason: 'Expected the app not to close after showing the toast',
+      );
+      await tester.binding.delayed(const Duration(seconds: 5));
+      expect(
+        await appObserver.haveCloseRequest(
+          () => toastObserver.expectShowsToast(
+            BackButtonInterceptor.popRoute,
+            message: l10n.pressOnceAgainToExit,
+            reason: 'Expected the app to ask for confirmation before exiting after the previous message timed out',
+          ),
+        ),
+        isFalse,
```

Comment:
also, for booleans prefer use `false` and `true`, these matchers are essentially the same

Reply:
I also tested if using `equals` is better, but the output is almost the same, there is no real value added by it, so I decided to not use it for the toast string comparisons.